### ISSUE TITLE
FURB171 Does not trigger when calling set(...)

### ISSUE THREAD

**Issue by naslundx** [astral-sh/ruff#15792]
### Description

Minimal code snippet:

```python
if 1 in {1}:  # This triggers
    print("Single-element set")

if 1 in set([1]):  # This does not, but is equivalent
    print("Single-element set")
```

Command:
`ruff check --isolated testfile.py --select FURB171 --preview`

No other settings. 

Ruff version 0.9.3.

**Comment by naslundx** [comment]
Sorry about that. Edited the original post, it should be `set([1])`. Indeed:

```python
>>> set([1]) == {1}
True
```

**Comment by dylwil3** [comment]
I'm not sure if the scope of the rule should be extended to include _composite_ constructions of singleton collections. There would be a lot of special cases to consider and it's not clear to me how one would draw the line for what should or shouldn't be considered. Already in this case if we add `set([1])` shouldn't we also add `set("a")` or `set((1,))`

**Comment by dylwil3** [comment]
> I'm not sure if the scope of the rule should be extended to include _composite_ constructions of singleton collections. There would be a lot of special cases to consider and it's not clear to me how one would draw the line for what should or shouldn't be considered. Already in this case if we add `set([1])` shouldn't we also add `set("a")` or `set((1,))`

@AlexWaygood has convinced me that this extension in scope has more pros than cons - thanks for suggesting it, and sorry for  delaying things a bit!
